worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -629,10 +629,17 @@ public void sort() throws InterruptedException {
           for (ShuffleBlockInfo blockInfo : originShuffleBlocks) {
             long offset = blockInfo.offset;
             long length = blockInfo.length;
-            ShuffleBlockInfo sortedBlock = new ShuffleBlockInfo();
-            sortedBlock.offset = fileIndex;
-            sortedBlock.length = length;
-            sortedShuffleBlocks.add(sortedBlock);
+            // combine multiple `ShuffleBlockInfo` into a single `ShuffleBlockInfo` of size
+            // less than `shuffleChunkSize`
+            if (!sortedShuffleBlocks.isEmpty() &&
+                sortedShuffleBlocks.get(sortedShuffleBlocks.size() - 1).length + length <= shuffleChunkSize) {
```

Comment:
@waitinfuture Just wanted to double check if `<=` condition would be fine here, or `shuffleChunkSize` is exclusive upperBound.

Reply:
`<=` is fine here, `shuffleChunkSize` is not exclusive upper bound, in fact it's not a strict bound.